pkg/gatherers/conditional/gather_containers_logs.go

```diff
@@ -57,37 +63,29 @@ func (g *Gatherer) gatherContainersLogs(
 	var records []record.Record
 
 	for _, alertLabels := range alertInstances {
-		podNamespace, err := getAlertPodNamespace(alertLabels)
+		info, err := parseAlertLabels(alertLabels)
 		if err != nil {
 			klog.Warningf(logMissingAlert, err.Error(), params.AlertName)
 			errs = append(errs, err)
-			continue
 		}
-		podName, err := getAlertPodName(alertLabels)
-		if err != nil {
-			klog.Warningf(logMissingAlert, err.Error(), params.AlertName)
-			errs = append(errs, err)
+		if len(info.namespace) == 0 {
 			continue
 		}
-		var podContainer string
-		if len(params.Container) > 0 {
-			podContainer = params.Container
-		} else {
-			podContainer, err = getAlertPodContainer(alertLabels)
-			if err != nil {
-				klog.Warningf(logMissingAlert, err.Error(), params.AlertName)
-				errs = append(errs, err)
-			}
-		}
 
 		logContainersFilter := common.LogContainersFilter{
-			Namespace:     podNamespace,
-			FieldSelector: fmt.Sprintf("metadata.name=%s", podName),
+			Namespace: info.namespace,
 		}
 
-		// The container label may not be present for all alerts (e.g., KubePodNotReady).
-		if len(podContainer) > 0 {
-			logContainersFilter.ContainerNameRegexFilter = fmt.Sprintf("^%s$", podContainer)
+		if len(params.Container) > 0 {
+			logContainersFilter.ContainerNameRegexFilter = fmt.Sprintf("^%s$", params.Container)
+		} else if len(info.container) > 0 {
+			logContainersFilter.ContainerNameRegexFilter = fmt.Sprintf("^%s$", info.container)
+		}
+
+		if len(params.PodName) > 0 {
+			logContainersFilter.PodNameRegexFilter = fmt.Sprintf("^%s$", params.PodName)
+		} else {
+			logContainersFilter.FieldSelector = fmt.Sprintf("metadata.name=%s", info.name)
```

Comment:
I would set the "defaults" in the `logContainersFilter` and remove the else here and above.

Reply:
We actually can't do this. There are somethings to be considered.

* For pod name:

If we make `FieldSelector` default, we will have to remove it inside the if, 'cause otherwise we are going to request to the API just the pod that is firing the alert, or saying that we want `metadata.name=`.

* For container:

Check this: https://github.com/openshift/insights-operator/pull/675#discussion_r970870885

Reply:
OK I see. Good point.